File: test/corpus/component/05-all-style-methods/case.ts
import { newOutput, Profile, rgbColor, withProfile } from '../../../../src/index.js';

// Force TrueColor profile for consistent testing
const out = newOutput(process.stdout, withProfile(Profile.TrueColor));

console.log('--- All Style Methods Test ---');

// Test basic text styling methods
const bold = out.string('Bold Text').bold();
const faint = out.string('Faint Text').faint();
const italic = out.string('Italic Text').italic();
const underline = out.string('Underlined Text').underline();
const overline = out.string('Overlined Text').overline();

console.log(`Bold: ${bold.toString()}`);
console.log(`Faint: ${faint.toString()}`);
console.log(`Italic: ${italic.toString()}`);
console.log(`Underline: ${underline.toString()}`);
console.log(`Overline: ${overline.toString()}`);

// Test special effects
const blink = out.string('Blinking Text').blink();
const reverse = out.string('Reversed Text').reverse();
const crossOut = out.string('Crossed Out Text').crossOut();

console.log(`Blink: ${blink.toString()}`);
console.log(`Reverse: ${reverse.toString()}`);
console.log(`CrossOut: ${crossOut.toString()}`);

// Test color combinations with styles
const styledColor = out.string('Bold Red Text').bold().foreground(rgbColor('#FF0000'));
const styledBg = out.string('Italic Blue Background').italic().background(rgbColor('#0000FF'));

console.log(`Styled Color: ${styledColor.toString()}`);
console.log(`Styled Background: ${styledBg.toString()}`);

// Test complex combinations
const complex = out
  .string('Bold Italic Underlined Red on Blue')
  .bold()
  .italic()
  .underline()
  .foreground(rgbColor('#FF0000'))
  .background(rgbColor('#0000FF'));

console.log(`Complex: ${complex.toString()}`);

File: test/corpus/component/05-all-style-methods/case.go
package main

import (
	"fmt"

	"github.com/muesli/termenv"
)

func main() {
	// Force TrueColor profile for consistent testing
	out := termenv.NewOutput(nil, termenv.WithProfile(termenv.TrueColor))
	
	fmt.Println("--- All Style Methods Test ---")
	
	// Test basic text styling methods
	bold := out.String("Bold Text").Bold()
	faint := out.String("Faint Text").Faint()
	italic := out.String("Italic Text").Italic()
	underline := out.String("Underlined Text").Underline()
	overline := out.String("Overlined Text").Overline()
	
	fmt.Printf("Bold: %s\n", bold)
	fmt.Printf("Faint: %s\n", faint)
	fmt.Printf("Italic: %s\n", italic)
	fmt.Printf("Underline: %s\n", underline)
	fmt.Printf("Overline: %s\n", overline)
	
	// Test special effects
	blink := out.String("Blinking Text").Blink()
	reverse := out.String("Reversed Text").Reverse()
	crossOut := out.String("Crossed Out Text").CrossOut()
	
	fmt.Printf("Blink: %s\n", blink)
	fmt.Printf("Reverse: %s\n", reverse)
	fmt.Printf("CrossOut: %s\n", crossOut)
	
	// Test color combinations with styles
	styledColor := out.String("Bold Red Text").Bold().Foreground(termenv.RGBColor("#FF0000"))
	styledBg := out.String("Italic Blue Background").Italic().Background(termenv.RGBColor("#0000FF"))
	
	fmt.Printf("Styled Color: %s\n", styledColor)
	fmt.Printf("Styled Background: %s\n", styledBg)
	
	// Test complex combinations
	complex := out.String("Bold Italic Underlined Red on Blue").
		Bold().
		Italic().
		Underline().
		Foreground(termenv.RGBColor("#FF0000")).
		Background(termenv.RGBColor("#0000FF"))
	
	fmt.Printf("Complex: %s\n", complex)
}